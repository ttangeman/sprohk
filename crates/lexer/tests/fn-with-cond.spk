// Compile-time restricted
const int: i32 = 42;

fn foo() -> ?i32 {
    // Constant
    let a: bool = true;
    // Variable
    var b = false;
    if a and b {
        return int;
    } else if a or b {
        return null;
    }
}