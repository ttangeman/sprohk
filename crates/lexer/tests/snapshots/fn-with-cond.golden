Token { kind: Keyword, loc: SourceLocation { line: 1, start: 27, end: 32 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 33, end: 36 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 36, end: 37 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 38, end: 41 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 42, end: 43 } }
Token { kind: NumberLiteral, loc: SourceLocation { line: 1, start: 44, end: 46 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 46, end: 47 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 49, end: 51 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 52, end: 55 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 55, end: 56 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 56, end: 57 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 58, end: 60 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 61, end: 62 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 62, end: 65 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 66, end: 67 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 88, end: 91 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 92, end: 93 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 93, end: 94 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 95, end: 99 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 100, end: 101 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 102, end: 106 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 106, end: 107 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 128, end: 131 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 132, end: 133 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 134, end: 135 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 136, end: 141 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 141, end: 142 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 147, end: 149 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 150, end: 151 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 152, end: 155 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 156, end: 157 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 158, end: 159 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 168, end: 174 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 175, end: 178 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 178, end: 179 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 184, end: 185 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 186, end: 190 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 191, end: 193 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 194, end: 195 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 196, end: 198 } }
Token { kind: Identifier, loc: SourceLocation { line: 1, start: 199, end: 200 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 201, end: 202 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 211, end: 217 } }
Token { kind: Keyword, loc: SourceLocation { line: 1, start: 218, end: 222 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 222, end: 223 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 228, end: 229 } }
Token { kind: Symbol, loc: SourceLocation { line: 1, start: 230, end: 231 } }